python/GafferArnoldUI/ArnoldShaderUI.py

```diff
@@ -178,9 +178,15 @@ def __translateNodeMetadata( nodeEntry ) :
 			__metadata[paramPath]["layout:section"] = page
 
 		# Label from OSL "label"
-		label = __aiMetadataGetStr( nodeEntry, paramName, "label", label )
+		label = __aiMetadataGetStr( nodeEntry, paramName, "label" )
 		if label is not None :
 			__metadata[paramPath]["label"] = label
+		elif "_" in paramName:
+			# Label from Arnold naming convention
+			# Arnold uses snake_case rather than camelCase for naming, so translate this into
+			# nice looking names
+			__metadata[paramPath]["label"] = " ".join( [ i.capitalize() for i in paramName.split( "_" ) ] )
```

Comment:
What happens to names without "_" in them? It seems like they'll remain uncapitalised?

Reply:
For names with a single token, they match Gaffer's camelCase naming convention, and Gaffer's autogenerated labels will be correct anyway.  Do you want me to generate labels for them as well for consistency, even though they won't change anything?

Reply:
Ah, of course. No, that's fine - I'd just missed the obvious.